### PR TITLE
feat(cli/js/testing): Add TestDefinition::only

### LINKED ISSUE #5197: Add "skip" and "only" functions on Deno.test
Improve Deno testing experience with "skip" and "only" features

## skip

Adding "skip" to the test would cause the test to be skipped when the tests are run with `deno test`
```ts
Deno.test.skip('skipped test', () => {
  // ...
});
```

https://jestjs.io/docs/en/api.html#testskipname-fn
https://mochajs.org/#inclusive-tests

## only

Adding "only" to the test would cause only tests with "only" to be ran when running `deno test`
```ts
Deno.test.only('only test example', () => {
  // ...
});
```

https://jestjs.io/docs/en/api.html#testonlyname-fn-timeout
https://mochajs.org/#exclusive-tests

## info

```
deno --version
deno 1.0.0-rc1
v8 8.2.308
```

### PR DESCRIPTION
**Pull request by nayeemrmn**
Closes #5197.

If there are tests with `only: true`, only those tests will be run and the runner will exit with code 1.

### CODE REVIEW

**Comment by bartlomieju** [comment]
@ry was this PR landed for `v1.1`?

**Comment by ry** [comment]
I'm not sure why this got closed... no I don't think it was landed.

File: cli/tests/deno_test_only.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Deno.test({
+  name: "abc",
+  fn() {},
+});
+
+Deno.test({
+  only: true,
+  name: "def",
+  fn() {},
+});
```
Comment (bartlomieju) [inline cli/tests/deno_test_only.ts:10]: Out of curiosity what happens when you do:
```
Deno.test({
  only: true,
  ignore: true,
  name: "def",
  fn() {},
});
```

Comment (nayeemrmn) [reply cli/tests/deno_test_only.ts:10]:
```
running 1 tests
test def ... ignored (0ms)

test result: ok. 0 passed; 0 failed; 1 ignored; 0 measured; 0 filtered out (1ms)

FAILED because the "only" option was used
```
First the `only` filter is applied (without affecting a metric), then the `--filter` filter is applied (increases the `filtered out` metric), then the `ignore` filter is applied (increases the `ignored` metric and is listed as `ignored`).

**Comment by ry** [comment]
Isn't this what the `deno test --filter` is for? It's confusing to me that "only" and "filter" are different words but have the same use...

**Comment by nayeemrmn** [comment]
@ry I don't think the name difference is a problem. One is a global filter and one is declaring exclusiveness for some test. The `only` option always overlaps with some CLI flag but is really useful practically. Not having to change the test command allows you to use the same `tools/test.sh`, which is especially important to have in Deno when your command may very well look like `deno test --allow-read --allow-env --allow-net -c tsconfig.json src`.

File: cli/tests/unit/test_util.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
 
 import { assert, assertEquals } from "../../../std/testing/asserts.ts";
+import * as colors from "../../../std/fmt/colors.ts";
+export { colors };
```
Comment (ry) [inline cli/tests/unit/test_util.ts:5]: Seems like unrelated changes?

File: cli/tests/unit/unit_test_runner.ts [hunk 3]
```diff
@@ -225,6 +227,13 @@ async function masterRunnerMain(
   }
 
   console.log("Unit tests passed");
+
+  if (REGISTERED_UNIT_TESTS.find(({ only }) => only)) {
+    console.error(
+      `\n${colors.red("FAILED")} because the "only" option was used`
+    );
+    Deno.exit(1);
+  }
```
Comment (ry) [inline cli/tests/unit/unit_test_runner.ts:236]: Does "deno test" also fail when only is used?

File: cli/js/lib.deno.ns.d.ts [hunk 4]
```diff
@@ -38,6 +38,7 @@ declare namespace Deno {
     fn: () => void | Promise<void>;
     name: string;
     ignore?: boolean;
+    only?: boolean;
```
Comment (ry) [inline cli/js/lib.deno.ns.d.ts:41]: Add JSDoc?

File: cli/tests/unit/test_util.ts (hunk 2 above)
Comment (nayeemrmn) [reply cli/tests/unit/test_util.ts:5]: It's re-exported for `unit_test_runner.ts` (the `export * as ...` syntax actually isn't standard yet).

File: cli/tests/unit/unit_test_runner.ts (hunk 3 above)
Comment (nayeemrmn) [reply cli/tests/unit/unit_test_runner.ts:236]: Yes, see test.

Comment (ry) [reply cli/tests/unit/unit_test_runner.ts:236]: So why is exit(1) necessary here?

Comment (nayeemrmn) [reply cli/tests/unit/unit_test_runner.ts:236]: The unit test runner doesn't use `deno test`'s `exitOnFail` because it needs to print summaries etc. Same reason there is another `Deno.exit(1)` above a bit.
